pkg/controllers/pod/pod_controller.go

```diff
@@ -174,12 +181,32 @@ func (r *PodReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 			r.StatusUpdater.UpdateFail(ctx, pod, err, "", setPodReadyStatusFalse)
 			return common.ResultRequeue, err
 		}
-		_, err = r.SubnetPortService.CreateOrUpdateSubnetPort(pod, nsxSubnet, contextID, &pod.ObjectMeta.Labels, false, false)
+		nsxSubnetPortState, err := r.SubnetPortService.CreateOrUpdateSubnetPort(pod, nsxSubnet, contextID, &pod.ObjectMeta.Labels, false, r.restoreMode)
 		if err != nil {
 			r.StatusUpdater.UpdateFail(ctx, pod, err, "", setPodReadyStatusFalse)
 			return common.ResultRequeue, err
 		}
+		if nsxSubnetPortState != nil && len(nsxSubnetPortState.RealizedBindings) > 0 &&
+			nsxSubnetPortState.RealizedBindings[0].Binding != nil &&
+			nsxSubnetPortState.RealizedBindings[0].Binding.MacAddress != nil {
+			podAnnotationChanges := map[string]string{
+				servicecommon.AnnotationPodMAC: strings.Trim(*nsxSubnetPortState.RealizedBindings[0].Binding.MacAddress, "\""),
+			}
+			err = util.UpdateK8sResourceAnnotation(r.Client, ctx, pod, podAnnotationChanges)
+			if err != nil {
+				log.Error(err, "Failed to update Pod annotation", "pod.Name", req.NamespacedName, "pod.UID", pod.UID, "podAnnotationChanges", podAnnotationChanges)
+				return common.ResultNormal, err
+			}
+		}
 		r.StatusUpdater.UpdateSuccess(ctx, pod, setPodReadyStatusTrue)
+		if r.restoreMode {
+			// Add restore annotation on Pod to notity Spherelet
+			retry.OnError(util.K8sClientRetry, func(err error) bool {
+				return err != nil
+			}, func() error {
+				return common.UpdateRestoreAnnotation(r.Client, ctx, pod, "eth0")
```

Comment:
I'm thinking maybe we first add annotation key before PodVM supports multi-nic

Reply:
I replaced eth0 by true for better readable, please let me know if you prefer other value. Thanks!